src/NuGet.Clients/NuGet.PackageManagement.UI/PackageItemLoader.cs

```diff
@@ -263,85 +263,101 @@ public IEnumerable<PackageItemViewModel> GetCurrent()
 
             foreach (PackageSearchMetadataContextInfo metadataContextInfo in _state.Results.PackageSearchItems)
             {
-                VersionRange allowedVersions = VersionRange.All;
-                VersionRange versionOverride = null;
+                var packageId = metadataContextInfo.Identity.Id;
+                var packageVersion = metadataContextInfo.Identity.Version;
 
-                // get the allowed version range and pass it to package item view model to choose the latest version based on that
-                if (_packageReferences != null)
+                if (listItemViewModels.TryGetValue(packageId, out PackageItemViewModel existingListItem))
                 {
-                    IEnumerable<IPackageReferenceContextInfo> matchedPackageReferences = _packageReferences.Where(r => StringComparer.OrdinalIgnoreCase.Equals(r.Identity.Id, metadataContextInfo.Identity.Id));
-                    VersionRange[] allowedVersionsRange = matchedPackageReferences.Select(r => r.AllowedVersions).Where(v => v != null).ToArray();
-                    VersionRange[] versionOverrides = matchedPackageReferences.Select(r => r.VersionOverride).Where(v => v != null).ToArray();
+                    existingListItem.InstalledVersions = existingListItem.InstalledVersions.Append(packageVersion);
+                }
+                else
+                {
+                    VersionRange allowedVersions = VersionRange.All;
+                    VersionRange versionOverride = null;
 
-                    if (allowedVersionsRange.Length > 0)
+                    // get the allowed version range and pass it to package item view model to choose the latest version based on that
+                    if (_packageReferences != null)
                     {
-                        allowedVersions = allowedVersionsRange[0];
+                        IEnumerable<IPackageReferenceContextInfo> matchedPackageReferences = _packageReferences.Where(r => StringComparer.OrdinalIgnoreCase.Equals(r.Identity.Id, metadataContextInfo.Identity.Id));
+                        var allowedVersionsRange = new List<VersionRange>();
+                        var versionOverrides = new List<VersionRange>();
+
+                        foreach (var reference in matchedPackageReferences)
+                        {
+                            if (reference.AllowedVersions != null)
+                            {
+                                allowedVersionsRange.Add(reference.AllowedVersions);
+                            }
+                            if (reference.VersionOverride != null)
+                            {
+                                versionOverrides.Add(reference.VersionOverride);
+                            }
+                        }
+
+                        allowedVersions = allowedVersionsRange.FirstOrDefault() ?? VersionRange.All;
+                        versionOverride = versionOverrides.FirstOrDefault();
```

Comment:
I believe we cannot move this to be outside the if, `allowedVersionsRange` and `versionsOverrides` are local variables inside this if, we could move those to be outside but since they are lists I think it's better to have 2 `VersionRanges` initialized. 

Also, since `versionOverrides` is initialized as an empty, `FirstOrDefault()` should return null if there are no values.